plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -488,6 +560,91 @@ fu_cros_ec_usb_device_send_done (FuDevice *device)
 	}
 }
 
+static gboolean
+fu_cros_ec_usb_device_send_subcommand (FuDevice *device, guint16 subcommand,
+				       gpointer cmd_body, gsize body_size,
+				       gpointer resp, gsize *resp_size,
+				       gboolean allow_less, GError **error)
+{
+	fu_cros_ec_usb_device_send_done (device);
+
+	if (!fu_cros_ec_usb_ext_cmd (device, subcommand,
+				     cmd_body, body_size,
+				     resp, resp_size, FALSE, error)) {
+		/* TODO: Insert prefix */
+		return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+
+static gboolean
+fu_cros_ec_usb_device_reset_to_ro (FuDevice *device, GError **error)
+{
+	gsize response_size;
+	guint8 response;
+	guint16 subcommand;
+	guint8 command_body[2]; /* Max command body size. */
+	gsize command_body_size;
+
+
+	command_body_size = 0;
+	response_size =1;
+
+	subcommand = UPDATE_EXTRA_CMD_IMMEDIATE_RESET;
+	if (!fu_cros_ec_usb_device_send_subcommand  (device, subcommand, command_body,
+				     command_body_size, &response,
+				     &response_size, FALSE, error)) {
+		g_print ("failed send subcommand\n");
+
+		/* failure here is ok */
+		g_clear_error (error);
+		return TRUE;
+	}
+
+	/* Send subcommand to remain in RO */
+/*	subcommand = UPDATE_EXTRA_CMD_STAY_IN_RO;
+	if (!fu_cros_ec_usb_device_send_subcommand (device, subcommand, command_body,
+				     command_body_size, &response,
+				     &response_size, FALSE, error))
+		return FALSE;
+*/
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_cros_ec_usb_device_jump_to_rw (FuDevice *device, GError **error)
+{
+	gsize response_size;
+	guint8 response;
+	guint16 subcommand;
+	guint8 command_body[2]; /* Max command body size. */
+	gsize command_body_size;
+
+
+	command_body_size = 0;
+	response_size =1;
+
+	subcommand = UPDATE_EXTRA_CMD_JUMP_TO_RW;
+	if (!fu_cros_ec_usb_device_send_subcommand  (device, subcommand, command_body,
```

Comment:
It looks like here the only difference is RO and RW use a different subcommand.  Also maybe can collapse these two and just have a single jump function with an argument where to jump

Reply:
The RO is now a sequence of two separate subcommands, once to tell the device to stay in ro, and then to initiate the immediate reset.

I think it might make sense to keep these separate for now.